src/jwt_authorizer/config.py

```diff
@@ -2,140 +2,300 @@
+
+    no_authorize: bool
+    """Disable authorization."""
+
+    no_verify: bool
+    """Disable token verification."""
+
+    set_user_headers: bool
+    """Whether to set headers containing user information from the token."""
+
+    username_key: str
+    """Token field from which to take the username."""
+
+    uid_key: str
+    """Token field from which to take the UID."""
+
+    group_mapping: Dict[str, List[str]]
+    """Mapping of a scope to a list of groups receiving that scope."""
+
+    issuer: IssuerConfig
+    """Configuration for internally-issued tokens."""
+
+    session_store: SessionStoreConfig
+    """Configuration for storing oauth2_proxy sessions."""
+
+    known_capabilities: Dict[str, str]
+    """Known scopes (the keys) and their descriptions (the values)."""
+
+    issuers: Dict[str, Issuer]
+    """Known iss (issuer) values and their metadata."""
+
+    @classmethod
+    def from_dynaconf(cls, settings: LazySettings) -> Config:
+        """Construction a Config object from Dynaconf settings.
 
         Parameters
         ----------
-        app : `flask.Flask`
-            The Flask application to configure.
-        user_config : `str`
-            An additional configuration file to load.
+        settings : `dynaconf.LazySettings`
+            Dynaconf settings.
+
+        Returns
+        -------
+        config : `Config`
+            The corresponding Config object.
         """
-        global logger
-        defaults_file = os.path.join(
-            os.path.dirname(__file__), "defaults.yaml"
+        if settings.get("OAUTH2_JWT.KEY"):
+            key = settings["OAUTH2_JWT.KEY"]
+        else:
+            key = cls._load_secret(settings["OAUTH2_JWT.KEY_FILE"])
+        issuer_config = IssuerConfig(
+            iss=settings["OAUTH2_JWT.ISS"],
+            kid=settings["OAUTH2_JWT.KEY_ID"],
+            aud=settings["OAUTH2_JWT.AUD.DEFAULT"],
+            aud_internal=settings["OAUTH2_JWT.AUD.INTERNAL"],
+            key=key,
+            exp_minutes=settings["OAUTH2_JWT_EXP"],
         )
 
-        settings_module = f"{defaults_file},{user_config}"
-        print(settings_module)
-        config = FlaskDynaconf(app, SETTINGS_FILE_FOR_DYNACONF=settings_module)
-        settings = config.settings
-        settings.validators.register(
-            Validator("NO_VERIFY", "NO_AUTHORIZE", is_type_of=bool),
-            Validator("GROUP_MAPPING", is_type_of=dict),
+        group_mapping = {}
+        if settings.get("GROUP_MAPPING"):
+            for key, value in settings["GROUP_MAPPING"].items():
+                assert isinstance(key, str), "group_mapping is malformed"
+                assert isinstance(value, list), "group_mapping is malformed"
+                group_mapping[key] = value
+
+        store_session_settings = settings["OAUTH2_STORE_SESSION"]
+        if store_session_settings.get("OAUTH2_PROXY_SECRET"):
+            secret_b64 = store_session_settings["OAUTH2_PROXY_SECRET"]
```

Comment:
Yeah, this is just for testing because creating a bunch of files in the test environment is kind of irritating. That's a good point about having one way to do it, though. I'll revisit this when it comes time to rewrite the configuration system to use pydantic, which will mean revisiting all the configuration decisions.